### PR TITLE
Add SplayTreeMap and Serf-Handler

### PR DESCRIPTION
**Pull request by wyhaines**
## Link
I am adding two of my shards with this PR.

https://github.com/wyhaines/splay_tree_map.cr
https://github.com/wyhaines/serf-handler.cr

## Checklist
* [x] - Shard is at least 30 days old.
* [x] - Shard has CI implemented.
* [x] - Shard has daily/weekly periodic builds (ideally with Crystal latest and nightly).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,6 +93,7 @@ Contributions are welcome. Please take a quick look at the [contribution guideli
  * [multiset.cr](https://github.com/tcrouch/multiset.cr) - Implementation of a multiset
  * [qr-code](https://github.com/spider-gazelle/qr-code) - QR Code generator
  * [radix](https://github.com/luislavena/radix) - Radix Tree implementation
+ * [splay_tree_map](https://github.com/wyhaines/splay_tree_map.cr) - Splay Tree implementation that conforms to the Hash ducktype.
```
Comment (Blacksmoke16) [inline README.md:96]:
```suggestion
 * [splay_tree_map](https://github.com/wyhaines/splay_tree_map.cr) - Splay Tree implementation that conforms to the Hash ducktype
```

To be consistent.

File: README.md [hunk 2]
```diff
@@ -332,6 +333,7 @@ Contributions are welcome. Please take a quick look at the [contribution guideli
  * [pinger](https://github.com/spider-gazelle/pinger) - Ping IP addresses and DNS entries without requiring sudo
  * [port_midi](https://github.com/jimm/crystal_port_midi) - Crystal C bindings for the PortMIDI cross-platform MIDI I/O library
  * [retriable.cr](https://github.com/Sija/retriable.cr) - Simple DSL to retry failed code blocks
+ * [serf-handler.cr](https://github.com/wyhaines/serf-handler.cr) - Framework for building Serf handlers, with a suite of useful builtin capabilities.
```
Comment (Blacksmoke16) [inline README.md:336]:
```suggestion
 * [serf-handler.cr](https://github.com/wyhaines/serf-handler.cr) - Framework for building Serf handlers, with a suite of useful builtin capabilities
```

Ditto.

**Review by Blacksmoke16 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (wyhaines) [reply README.md:96]: Ack. I didn't even notice the periods escaping my fingers.  I will update.  Thanks!

**Review by veelenga — APPROVED** [review]

**Review by Blacksmoke16 — APPROVED** [review]

**Comment by Blacksmoke16** [comment]
@wyhaines Mind taking care of that merge conflict?

**Comment by wyhaines** [comment]
Conflict has been resolved.

**Comment by Blacksmoke16** [comment]
@wyhaines Looks like there's still one more?

**Pull request merged**